src/test/java/teammates/test/cases/webapi/GetInstructorPrivilegeActionTest.java

```diff
@@ -20,16 +31,329 @@ protected String getRequestMethod() {
+        dataBundle.instructors.put("instructor1OfCourse1", instructor1ofCourse1);
+        removeAndRestoreDataBundle(dataBundle);
+    }
+
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        // see individual tests.
+    }
+
+    @Test
+    protected void testExecute_fetchPrivilegeOfNonExistInstructor() {
+        InstructorAttributes instructor1ofCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1ofCourse1.getGoogleId());
+        ______TS("Request to fetch privilege of another instructor");
+        // course id is used for instructor identify verification here.
+        String[] invalidInstructorParams = {
+                Const.ParamsNames.COURSE_ID, instructor1ofCourse1.getCourseId(),
+                Const.ParamsNames.INSTRUCTOR_ID, "invalidid",
+        };
+
+        GetInstructorPrivilegeAction a = getAction(invalidInstructorParams);
+        JsonResult result = getJsonResult(a);
+        MessageOutput output = (MessageOutput) result.getOutput();
+
+        assertEquals(HttpStatus.SC_NOT_FOUND, result.getStatusCode());
+        assertEquals("Instructor does not exist.", output.getMessage());
+    }
+
+    @Test
+    protected void testExecute_fetchPrivilegeOfAnotherInstructor() {
+        InstructorAttributes instructor1ofCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        InstructorAttributes instructor2ofCourse1 = dataBundle.instructors.get("instructor2OfCourse1");
+        loginAsInstructor(instructor1ofCourse1.getGoogleId());
+        ______TS("Request to fetch privilege of another instructor");
+        // course id is used for instructor identify verification here.
+        String[] anotherInstructorParams = {
+                Const.ParamsNames.COURSE_ID, instructor1ofCourse1.getCourseId(),
+                Const.ParamsNames.INSTRUCTOR_ID, instructor2ofCourse1.getGoogleId(),
+        };
+
+        GetInstructorPrivilegeAction a = getAction(anotherInstructorParams);
+        InstructorPrivilegeData response = (InstructorPrivilegeData) getJsonResult(a).getOutput();
+
+        assertFalse(response.isCanModifyCourse());
+        assertTrue(response.isCanModifyStudent());
+        assertTrue(response.isCanModifyInstructor());
+        assertTrue(response.isCanModifySession());
+
+        assertTrue(response.isCanViewStudentInSections());
+
+        assertTrue(response.isCanModifySessionCommentsInSections());
+        assertTrue(response.isCanViewSessionInSections());
+        assertTrue(response.isCanSubmitSessionInSections());
+    }
+
+    @Test
+    protected void testExecute_notEnoughParameters() {
+        InstructorAttributes instructor1ofCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1ofCourse1.getGoogleId());
+        ______TS("Not enough parameters");
+        verifyHttpParameterFailure();
+    }
+
+    @Test
+    protected void testExecute_withCourseId() {
```

Comment:
How about `withInvaidCourseId`?

Reply:
That will be the scope of accessControl?

```
    public void checkSpecificAccessControl() {
        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
        if (instructor == null) {
            throw new UnauthorizedAccessException("Not instructor of the course");
        }
    }
```